src/shop.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  ShopConfiguartion,
+  Product,
+  CheckoutOptions,
+  Order,
+} from './types/shop'
+import { Storage } from './storage'
+import HubApi from '@nimiq/hub-api'
+
+export class Shop {
+  private configuration: ShopConfiguartion
+  private hubApi: HubApi
+  private storage: Storage
+
+  constructor(configuration: ShopConfiguartion) {
+    const { hubUrl, id, live } = configuration
+
+    this.configuration = configuration
+    this.storage = new Storage(id, live, live)
+    this.hubApi = new HubApi(
+      hubUrl || `https://hub.nimiq${live ? '' : '-testnet'}.com`,
+    )
+  }
+
+  // TODO(svub) Q: Meta data is to be freely defined by shop owner. String or Object? (will be run through JSON.stringify anyhow)
+  async checkout(products: Product[], meta: string): Promise<string> {
```

Comment:
If we want to give the shop user all the freedom, then `meta` might be a `number` or `boolean` or anything else. So `any` would be fine then (as you suggested anywhere before). 

Another idea: As soon as `Json` type will made it into a next version of `TypeScript` (`3.7` or so, see https://github.com/Microsoft/TypeScript/issues/1897), we could force the user to use JSON as `meta` only. 

Also I would recommend to make `meta` optional: `meta?: Json` or `meta?: any` (for now)

Reply:
optional, sure!
Json also good with me